.github/workflows/linux-sampleapp-ci.yml

```diff
@@ -0,0 +1,76 @@
+name: Linux SampleApp CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install system dependencies
+      run: |
+        # build-essential clang-11 libc++-dev libc++abi-dev libc++1 libc++abi1 ruby-full cmake
+        sudo apt-get update && sudo apt-get install ninja-build llvm-11 libboost-all-dev clang-11 --fix-missing
+    - name: Cache internal dependencies
+      id: internal_cache
+      uses: actions/cache@v1.1.2
+      with:
+        path: build/local
+        key: ${{ runner.os }}-internal-${{ hashFiles('tools/install-dependencies') }}
+    - name: Install internal dependencies
+      run: |
+        tools/install-dependencies
+      env:
+        CC: /usr/bin/clang
+        CXX: /usr/bin/clang++
+      if: steps.internal_cache.outputs.cache-hit != 'true'
+    - name: Code generation
+      run: |
+        tools/generate-files
+      env:
+        CC: /usr/bin/clang
+        CXX: /usr/bin/clang++
+    - name: CMake
+      run: |
+        cmake -H. -Bbuild -DCMAKE_BUILD_TYPE=Debug
+      env:
+        CC: /usr/bin/clang
+        CXX: /usr/bin/clang++
+    - name: Build
+      run: |
+        make -Cbuild -j12 TrustWalletCore
+      env:
+        CC: /usr/bin/clang
+        CXX: /usr/bin/clang++
+        CK_TIMEOUT_MULTIPLIER: 4
+    - name: Build and run C++ sample app
+      run: |
+        cd samples/cpp
+        cmake . -DWALLET_CORE=../../ -DCMAKE_BUILD_TYPE=Debug
+        make
+        ./sample || true
+      env:
+        CC: /usr/bin/clang
+        CXX: /usr/bin/clang++
+    - name: Install Go
```

Comment:
instead of installing this, is there a way to setup CI with build-in configuration of tools?

Reply:
See example with `matrix`: https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-running-with-multiple-operating-systems

Reply:
> instead of installing this, is there a way to setup CI with build-in configuration of tools?

Simplest is to use Golang version that comes with the Ubuntu v2004 available in the CI env (via apt-get), but unfortunately that version is too old, hence 'manual' install.